core/src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -174,7 +176,16 @@ public static void parseNumberField(NumberFieldMapper.Builder builder, String na
                 builder.omitNorms(nodeBooleanValue(propNode));
                 iterator.remove();
             } else if (propName.equals("similarity")) {
-                builder.similarity(parserContext.getSimilarity(propNode.toString()));
+                String value = propNode.toString();
+                if (parserContext.indexVersionCreated().before(Version.V_3_0_0) &&
+                    "default".equals(value) && parserContext.getSimilarity(value) == null) {
```

Comment:
Not relevant anymore, I though it was possible to override the default similarity with a custom one but I realized afterward that it was not possible. I'll remove it.